Cython/Utility/CythonFunction.c

```diff
@@ -367,37 +367,33 @@ __Pyx_CyFunction_get_annotations(__pyx_CyFunctionObject *op, CYTHON_UNUSED void
     return result;
 }
 
-#if PY_VERSION_HEX >= 0x030402C1
 static PyObject *
 __Pyx_CyFunction_get_is_coroutine(__pyx_CyFunctionObject *op, CYTHON_UNUSED void *context) {
-#if PY_VERSION_HEX >= 0x030503C1
+#if PY_VERSION_HEX >= 0x03050000
     // on v3.5.3rc1 the marker object was introduced
-    if (!op->func_is_coroutine) {
-        if (op->flags & __Pyx_CYFUNCTION_COROUTINE) {
-            PyObject *module, *fromlist, *marker = PYIDENT("_is_coroutine");
-            fromlist = PyList_New(1);
-            if (unlikely(!fromlist)) goto ignore;
-            PyList_SET_ITEM(fromlist, 0, marker);
-            module = PyImport_ImportModuleLevelObject(PYIDENT("asyncio.coroutines"), NULL, NULL, fromlist, 0);
-            Py_DECREF(fromlist);
-            if (unlikely(!module)) goto ignore;
-            op->func_is_coroutine = __Pyx_PyObject_GetAttrStr(module, marker);
-            Py_DECREF(module);
-            if (unlikely(!op->func_is_coroutine)) goto ignore;
-            return op->func_is_coroutine;
-        }
-    ignore:
-        op->func_is_coroutine = Py_None;
+    if (op->func_is_coroutine)
+        return __Pyx_NewRef(op->func_is_coroutine);
+    if (op->flags & __Pyx_CYFUNCTION_COROUTINE) {
+        PyObject *module, *fromlist, *marker = PYIDENT("_is_coroutine");
+        fromlist = PyList_New(1);
+        if (unlikely(!fromlist)) return NULL;
+        PyList_SET_ITEM(fromlist, 0, marker);
+        module = PyImport_ImportModuleLevelObject(PYIDENT("asyncio.coroutines"), NULL, NULL, fromlist, 0);
+        Py_DECREF(fromlist);
+        if (unlikely(!module)) goto ignore;
+        op->func_is_coroutine = __Pyx_PyObject_GetAttrStrNoError(module, marker);
+        Py_DECREF(module);
+        if (unlikely(!op->func_is_coroutine)) goto ignore;
+        return __Pyx_NewRef(op->func_is_coroutine);
     }
-    Py_INCREF(op->func_is_coroutine);
-    return op->func_is_coroutine;
+ignore:
+    op->func_is_coroutine = Py_None;
+    return __Pyx_NewRef(op->func_is_coroutine);
 #else
     PyObject *result = (op->flags & __Pyx_CYFUNCTION_COROUTINE) ? Py_True : Py_False;
-    Py_INCREF(result);
-    return result;
+    return __Py_NewRef(result);
```

Comment:
There's even a function for this already: `__Pyx_PyBool_FromLong()`.

Reply:
Good to know... Using it.